ts/ui/lazy/LazyHandler.ts

```diff
@@ -0,0 +1,483 @@
+ * @template B  The MathDocument class to extend
+ */
+export function LazyMathDocumentMixin<N, T, D,
+B extends MathDocumentConstructor<HTMLDocument<N, T, D>>>(
+  BaseDocument: B
+): MathDocumentConstructor<HTMLDocument<N, T, D>> & B {
+
+  return class BaseClass extends BaseDocument {
+
+    /**
+     * The Intersection Observer used to track the appearance of the expression markers
+     */
+    public lazyObserver: IntersectionObserver;
+
+    /**
+     * The mapping of markers to MathItems
+     */
+    public lazyList: LazyList<N, T, D>;
+
+    /**
+     * A promise to make sure our compiling/typesetting is sequential
+     */
+    protected lazyPromise: Promise<void> = Promise.resolve();
+
+    /**
+     * The function used to typeset a set of lazy MathItems.
+     *   (uses requestIdleCallback if available, or setTimeout otherwise)
+     */
+    protected lazyProcessSet: () => void;
+
+    /**
+     * True when a set of MathItems is queued for being processed
+     */
+    protected lazyIdle: boolean = false;
+
+    /**
+     * The set of items that have come into view
+     */
+    protected lazySet: LazySet = new Set();
+
+    /**
+     * Augment the MathItem class used for this MathDocument,
+     *   then create the intersection observer and lazy list,
+     *   and bind the lazyProcessSet function to this instance
+     *   so it can be used as a callback more easily.
+     *
+     * @override
+     * @constructor
+     */
+    constructor(...args: any[]) {
+      super(...args);
+      this.options.MathItem =
+        LazyMathItemMixin<N, T, D, Constructor<HTMLMathItem<N, T, D>>>(this.options.MathItem);
+      this.lazyObserver = new IntersectionObserver(this.lazyObserve.bind(this));
+      this.lazyList = new LazyList<N, T, D>();
+      const callback = this.lazyHandleSet.bind(this);
+      this.lazyProcessSet = (typeof window !== 'undefined' && window.requestIdleCallback ?
+                          () => window.requestIdleCallback(callback) :
+                          () => setTimeout(callback, 10));
+    }
+
+    /**
+     * The function used by the IntersectionObserver to monitor the markers coming into view.
+     * When one (or more) does, add it to or remove it from the set to be processed, and
+     *   if added to the set, queue an idle task, if one isn't already pending.
+     *
+     * @param {IntersectionObserverEntry[]} entries   The markers that have come into or out of view.
+     */
+    protected lazyObserve(entries: IntersectionObserverEntry[]) {
+      for (const entry of entries) {
+        const id = this.adaptor.getAttribute(entry.target as any as N, LAZYID);
+        const math = this.lazyList.get(id);
+        if (!math) return;
+        if (entry.isIntersecting) {
+          this.lazySet.add(id);
+          if (!this.lazyIdle) {
+            this.lazyIdle = true;
+            this.lazyProcessSet();
+          }
+        } else {
+          this.lazySet.delete(id);
+        }
+      }
```

Comment:
Please simplify this `if-if-else`.

Reply:
I'm not sure what you have in mind (or what the objection is).  I have restructured using `break`, so perhaps that will do.